src/Microsoft.AspNetCore.Proxy/ProxyServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,26 @@
+﻿
+using System;
+using Microsoft.AspNetCore.Proxy;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    public static class ProxyServiceCollectionExtensions
+    {
+        public static IServiceCollection AddProxy(this IServiceCollection services)
+        {
+            return services.AddSingleton(new ProxyService(new SharedProxyOptions()));
```

Comment:
> AddSingleton<ProxyService>() and then ProxyService takes an IOptions<SharedProxyOptions> - 

If you really think it is a good idea we can go this way. My concern is that auto-injection is slightly slower, but I can adjust.